### DISCUSSION TITLE
using cache.updateQuery upon an existing query result instead of re-fetchinig again after having a mutation on it, React

### DISCUSSION THREAD

**Discussion by EssamKayal96** [urql-graphql/urql#2968] [Q&A] [answered]
I am working on a website using REACT.

The website is full of tables which have a lot of actions in, so instead of re-fetching the data after making a change I am seeking to alter the existing state that is mapped for the initial query.

I already read about updating the cache, but I found it kind of ambiguous to write in REACT, and I'v searched for any example of using it in REACT but I ended up with nothing, so kindly help me here with a simple example of how using it.

**Comment by kitten** [comment, marked as answer]
None of the cache updates code is written "in React"; in other words, the GraphQL client and all its parts (the exchanges) live and are configured outside of UI/React code. So, in your case, you're looking for how to configure Graphcache.

We have some docs on the various things you can do with it, and the docs on how to update it live here: https://formidable.com/open-source/urql/docs/graphcache/cache-updates/

Additionally, there's a 'minimal' example for Graphcache here: https://github.com/urql-graphql/urql/tree/main/examples/with-graphcache-updates

In general, you may not even need a Cache Update, or you won't need to use `cache.updateQuery` and can instead rely either on `cache.updateFragment` or a simple `cache.link`.
